### ISSUE TITLE
MASS: Masked Sequence to Sequence Pre-training for Language Generation

### ISSUE THREAD

**Issue by fhiyo** [fhiyo/ml_papers#41]
## 0. 論文

<https://arxiv.org/abs/1905.02450>

Kaitao Song * 1 Xu Tan * 2 Tao Qin 2 Jianfeng Lu 1 Tie-Yan Liu 2

\*Equal contribution
1 Key Laboratory of Intelligent Perception and Systems for High-Dimensional Information of Ministry of Education, Nanjing University of Science and Technology 
2 Microsoft Research

Accepted by ICML 2019

## 1. どんなもの？

encoder-decoderベースの言語生成のためのpre-trainingの新しい手法であるMASS (MAsked Sequence to Sequence pre-training) の提案。encoder側で入力の連続した部分系列にマスクをかけ、decoder側でencoder側でマスクをしていない部分にマスクをかけた上でencoder側でマスクをかけた部分系列を予測するというpre-trainを行う。このpre-trainを行ったモデルを用いて、(low-resource or unsupervisedな)機械翻訳、文章要約、対話文応答生成のタスクで評価したところSOTAを達成した。

翻訳タスクでは一般的に翻訳元-翻訳先の文のペアが学習データとして必要だが、多くの言語はそのようなペアのデータは無く、代わりにペアになっていないデータが大量にある状態である。そこで、単一の言語でsource側の表現を上手く抽出できるようにdecoder側を学習できるようなタスクで訓練させたモデルを構築できれば、low-resourceであったりunsupervisedな機械翻訳のタスクにも性能が出るであろう、と仮説を立てて構築したモデル？

### pre-training

![image](https://user-images.githubusercontent.com/13327262/67624059-408fda00-f867-11e9-9742-60e2ba781dba.png)

encoderとdecoderで同じ言語を用いて、encoder側で文の連続した一部分にマスクをかけてその表現を獲得し、decoder側でencoder側でマスクをかけていない部分にマスクをかけてencoder側でマスクした部分の予測を行う。機械翻訳のタスクにおいてはsourceとtargetの言語を用いてpre-trainingし、文章要約、対話文応答生成のタスクにおいては対象となる単一の言語を用いてpre-trainingした。

## 4. どうやって有効だと検証した？

MASSのpre-trainingしていない版のモデルと、BERT+LM, DAEのモデルをbaselineとして実験を行った。
fine-tuneに用いるデータの量を調節し、データ量が少ないときどのような性能が出るかを実験した。結果はbaselineと比較して高い性能が出た。

## 5. 議論はある？

BERT, standard language modelingは本手法の特殊なバージョンである、と解釈ができる。

![image](https://user-images.githubusercontent.com/13327262/67623861-31a82800-f865-11e9-9b7f-ce5d91734b8c.png)

BERTが解いている穴埋めのタスクはmaskのサイズが1の場合、(GPTなどの) standard LMが解いているタスクはsourceが全てマスクされている場合と見ることができ、本手法はその中間のような手法になる。

encoder側のマスクが占める割合であるparameter kを変化させたときにどのように性能が変化するかを実験している。

![image](https://user-images.githubusercontent.com/13327262/67624164-420dd200-f868-11e9-8716-a8ec3f12b0c2.png)

実験の結果として、50% - 70%くらいのマスクの割合のときに性能が最も高くなる事がわかった。これは、encoder側かdecoder側のどちらかに有効なトークンが少なすぎる場合、モデルが推論する根拠がどちらかに偏ることになる。生成タスクがencoder内の文表現を抽出し、decoder側が自身のstateとencoder側の表現を利用するという協調的なタスクを行っていることからこの現象は生成モデルを構築するのに望ましくないものである、と考えられる。一方マスクの割合を50%程度に調節した場合、BERTのように単一のtokenを予測する場合と比べ、decoder側はencoder側の表現からより上手く情報を抽出しなければ解けない問題を解くために生成タスクを行った際の性能も高くなるようだ。
